core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1371,16 +1371,21 @@ private Scope createScopeForCommonTableExpression(Table table, Optional<Scope> s
                 fields = fieldBuilder.build();
             }
             else {
-                fields = queryDescriptor.getAllFields().stream()
-                        .map(field -> Field.newQualified(
+                ImmutableList.Builder<Field> fieldBuilder = ImmutableList.builder();
```

Comment:
It should just work if you add `.filter(field -> !field.isHidden())` instead of rewriting this as a for loop.

Reply:
It says in javadoc for the `getAllFields()` method that it doesn't guarantee the order of columns.

Reply:
fwiw `getAllFields` returns a `Collection`, not a `List`, probably to reiterate what the javadoc is saying